flows/inference.py

```diff
@@ -570,18 +571,45 @@ async def classifier_inference(
     for classifier_spec in classifier_specs:
         batches = iterate_batch(validated_file_stems, batch_size)
 
-        tasks = [
-            run_deployment(
+        ttl_per_task_s = 20 * 60
+
+        async def run_and_wait_for_deployment(
+            flow_name, deployment_name, batch, config, classifier_spec
+        ):
+            flow_run: FlowRun = await run_deployment(
                 name=f"{flow_name}/{deployment_name}",
                 parameters={
                     "batch": batch,
                     "config_json": config.to_json(),
                     "classifier_name": classifier_spec.name,
                     "classifier_alias": classifier_spec.alias,
                 },
-                timeout=1200,
+                timeout=0,
```

Comment:
suggestion: When I saw this I thought ahh won't we just raise immediately. Checked the prefect package and it says: 

> Setting `timeout` to 0 will return the flow run metadata immediately.

Wondering whether it's worth chucking a comment above, no dramas though.

Reply:
Haha yup, I covered this in the beginning of the PR description, with a quote and link to the docs. I've got it as a comment in 3 other places, so I probably should add it here!